### PR TITLE
Fixing mysql role. Unifying Debian and RedHat code by means of conditional imports.

### PR DESCRIPTION
**Pull request by mbarcia**
Fix to #188, unifiying both OS families into one set of scripts.
Note that this PR includes commits in PR #197.

### CODE REVIEW

File: vlad/playbooks/site.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@
     - "./vars/defaults/general.yml"
```
Comment (mbarcia) [inline vlad/playbooks/site.yml]: From now on is PR #197

File: vlad/playbooks/roles/mysql/templates/mysql.my.cnf.j2 [hunk 2]
```diff
@@ -20,7 +20,7 @@ skip-external-locking
 
 innodb_buffer_pool_size={{ innodb_buffer_pool_size }}
 {% if innodb_file_per_table == true %}
-innodb-file-per-table
+innodb_file_per_table
```
Comment (mbarcia) [inline vlad/playbooks/roles/mysql/templates/mysql.my.cnf.j2]: Correct setting name.

File: vlad/playbooks/roles/mysql/tasks/mysql.yml [hunk 3]
```diff
@@ -53,13 +56,25 @@
...
 
+- name: force restart mysql
+  service: name=mysqld state=restarted
+  when: ansible_os_family == "RedHat"
+  sudo: true
+
 # MySQL database setup, this does the equivalent of mysql_secure_installation.
 - name: set the root password.
-  mysql_user: name=root password={{ mysql_root_password }} host=localhost
+  mysql_user:
```
Comment (mbarcia) [inline vlad/playbooks/roles/mysql/tasks/mysql.yml]: Main fix to #188

File: vlad/playbooks/roles/mysql/tasks/mysql.yml [hunk 4]
```diff
@@ -94,7 +109,7 @@
 # Optional database import
 
 - name: get current database state
-  shell: mysql {{ dbname[0] }} --execute="show tables;"
+  shell: mysql {{ dbname }} --execute="show tables;"
```
Comment (mbarcia) [inline vlad/playbooks/roles/mysql/tasks/mysql.yml]: This little bug is now squashed.

File: vlad/playbooks/roles/mysql/tasks/mysql.yml [hunk 5]
```diff
@@ -53,13 +56,25 @@
...
+    host=localhost
+    login_user=root 
+    login_password={{ mysql_root_password }}
+    check_implicit_admin=true 
+    priv=*.*:ALL,GRANT
 
 - name: config for easy access as root user
   template: src=mysql_root.my.cnf.j2 dest=/root/.my.cnf
   when: "'{{ user }}' == 'root'"
+  sudo: true
```
Comment (mbarcia) [inline vlad/playbooks/roles/mysql/tasks/mysql.yml]: Natural.

File: vlad/playbooks/roles/mysql/tasks/mysql.yml [hunk 6]
```diff
@@ -53,13 +56,25 @@
   notify:
    - save iptables
 
+- name: force restart mysql
```
Comment (mbarcia) [inline vlad/playbooks/roles/mysql/tasks/mysql.yml]: I didn't innovate here. Previous RedHat script required a forced restart.

File: vlad/playbooks/roles/mysql/tasks/mysql.yml [hunk 7]
```diff
@@ -2,38 +2,41 @@
...
+    src=mysql.my.cnf.j2
+    dest="{{ mysql_config_dir }}/my.cnf"
   sudo: true
   register: mysql_conf
   notify:
   - restart mysql
 
 - name: remove the InnoDB log files in case InnoDB settings have been changed (1/2).
-  file: path=/var/lib/mysql/ib_logfile0 state=absent
+  file: path="{{ mysql_install_dir }}/ib_logfile0 state=absent"
```
Comment (mbarcia) [inline vlad/playbooks/roles/mysql/tasks/mysql.yml]: This value is the same in both OS families, but it's better to get rid of magic paths, even paths like these.

**Comment by philipnorton42** [comment]
There are a lot of good changes in here, but as it contains the changes from the previous pull request (which I have just closed) then I'm going to close this as well.

**Pull request closed without merging**